cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIFabricSetting.scala

```diff
@@ -0,0 +1,206 @@
+// Tenant setting status check by model name ("GPT-4", "gpt-35-turbo")
+// eg: FabricTenant.getModelStatus("GPT-4")
+// returnType: Boolean
+
+// Part of the code are scala implementation of
+// synapse.ml.internal_utils.session_utils
+// synapse.ml.fabric.token_utils
+// synapse.ml.mlflow.synapse_mlflow_utils
+
+import spray.json._
+
+import scala.io.Source
+import scala.collection.immutable.Map
+import scala.util.Try
+import java.net.URI
+import java.net.InetAddress
+import StringifiedJsonProtocol._
+
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.client.methods.{HttpGet, HttpPost}
+import org.apache.http.entity.StringEntity
+import org.apache.http.util.EntityUtils
+
+
+object StringifiedJsonProtocol extends DefaultJsonProtocol {
+  implicit object StringifiedMapFormat extends RootJsonFormat[Map[String, String]] {
+    def write(map: Map[String, String]): JsValue = {
+      JsObject(map.mapValues(JsString(_)))
+    }
+
+    def read(value: JsValue): Map[String, String] = value.asJsObject.fields.map {
+      case (key, JsString(str)) => key -> str
+      case (key, other) => key -> other.toString
+    }
+  }
+}
+
+object ConfigConstants {
+  final val CONTEXT_FILE_PATH = "/home/trusted-service-user/.trident-context"
+  final val TOKEN_PATH = "/opt/token-service/tokenservice.config.json"
+  final val TRIDENT_LAKEHOUSE_TOKEN_SERVICE_ENDPOINT = "trident.lakehouse.tokenservice.endpoint"
+  final val TRIDENT_SESSION_TOKEN = "trident.session.token"
```

Comment:
Nit on Style: no camelcase in scala, you can just use the regular naming case (with a capital first letter because its an object